proposals/security_and_privacy.md

```diff
@@ -0,0 +1,277 @@
+current mechanisms (cookies, etc).
+
+WoT Things can also act as clients, however, accessing
+web services and other WoT Things acting as servers.
+It is also possible that the same device could instantiate
+multiple WoT Runtimes.  However, there is noi explicit mechanism
+for sharing state between such instances, and each instance
+can be considered a single origin.
+
+### Does this specification expose persistent, cross-origin state to the web? 
+
+A WoT Thing may act as a server and expose interactions that 
+allow properties to be modified.  Therefore modifications by
+one client may be visible to another.
+
+This risk of this, if any, is mitigated by the fact that
+WoT Things support various mechanisms to limit access to 
+authorized users, like any other web service.
+
+When acting as a client, each WoT Thing is in effect single-origin.
+
+There is a potential fingerprinting risk, which is discussed,
+and is related to the PII risk discussed above.
+This can be mitigated by controlling access to WoT metadata
+such as WoT Thing Descriptions.
+
+### Does this specification expose any other data to an origin that it doesn’t currently have access to?
+
+Yes.
+
+WoT Things by nature will be connected into a network of
+devices.  A single "WoT Network" application may involve
+communications between many WoT Things, some acting as
+servers, some acting as clients, some acting as both.
+
+In addition, WoT Things may act as gateways to other
+non-HTTP devices using other network protocols, such
+as MQTT or CoAP, or to interface to a variety of non-IP
+devices using BACNET, ECHONET, X10, SPI, I2C, and many
+other possibilities.
+
+Unfortunately, the single-origin model used in 
+web servers and browsers does not scale well to this model.
+
+In practice, securing an IoT Application in general is
+out of scope.  We do however discuss several common connection
+patterns (client-server, proxy, etc) and how they can be secured.
+
+This risk is mitigated by the fact that endpoint IoT devices
+are generally not exposed directly on the internet, but
+are mediated by gateways and cloud services.
+
+### Does this specification enable new script execution/loading mechanisms? 
+
+Yes.  It enables WoT Things to run applications based
+on ECMAScript, using a constrained API, and with suitable
+restrictions, such as no direct access to private
+security data.
+
+However, a mechanism for loading new scripts is not defined
+and is currently considered to be out of scope.
+Our architecture assumes the scripts running inside a WoT Thing
+are trusted implementations provided by the manufacturer
+of the device and installed using a suitable OTA update
+mechanism; but this update mechanism is not defined in our
+architecture and is also currently considered to be out of scope.
+
+### Does this specification allow an origin access to a user’s location?
+
+As noted above in the PII response, in theory a device
```

Comment:
Assigning a location to a device is common practice in industrial use cases. Suggest to remove "in theory".

Reply:
Hmm... let me think about it.   Maybe I should differentiate different use cases here: industrial, personal, and medical, which have different requirements.   Probably should include those different views in the main doc too.

Reply:
Hmm... note I was talking about "the user's location" not "the device's location".   The "in theory" part is because in some cases you can infer the user's location from the device's location.  But in some cases that may be intentional.  The privacy hazard is that you don't want *unintentional* disclosure of user location.

Reply:
I deleted the "in theory", and added a sentence stating that is in fact common in institutional use cases.